From 81b3a799a1a3ca88abeae9066df611269d42a36b Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 14 Jul 2014 13:07:53 -0400
Subject: [x86] therm_throt: Remove unused therm_cpu_lock

Message-id: <1405343292-21004-17-git-send-email-prarit@redhat.com>
Patchwork-id: 85763
O-Subject: [RHEL7.1 PATCH BZ 1119078 16/35] x86, therm_throt.c: Remove unused therm_cpu_lock
Bugzilla: 1119078
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1119078

commit 7b7139d4abe8a94446db6a91882f1400920891d5
Author: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
Date:   Tue Mar 11 02:07:28 2014 +0530

    x86, therm_throt.c: Remove unused therm_cpu_lock

    After fixing the CPU hotplug callback registration code, the callbacks
    invoked for each online CPU, during the initialization phase in
    thermal_throttle_init_device(), can no longer race with the actual CPU
    hotplug notifier callbacks (in thermal_throttle_cpu_callback). Hence the
    therm_cpu_lock is unnecessary now. Remove it.

    Cc: Tony Luck <tony.luck@intel.com>
    Cc: Borislav Petkov <bp@alien8.de>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Cc: Ingo Molnar <mingo@kernel.org>
    Cc: "H. Peter Anvin" <hpa@zytor.com>
    Suggested-by: Oleg Nesterov <oleg@redhat.com>
    Signed-off-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Tony Camuso <tcamuso@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/cpu/mcheck/therm_throt.c b/arch/x86/kernel/cpu/mcheck/therm_throt.c
index fe55e27..10a787b 100644
--- a/arch/x86/kernel/cpu/mcheck/therm_throt.c
+++ b/arch/x86/kernel/cpu/mcheck/therm_throt.c
@@ -253,9 +253,6 @@ static void thermal_throttle_remove_dev(struct device *dev)
  sysfs_remove_group(&dev->kobj, &thermal_attr_group);
 }
 
-/* Mutex protecting device creation against CPU hotplug: */
-static DEFINE_MUTEX(therm_cpu_lock);
-
 /* Get notified when a cpu comes on/off. Be hotplug friendly. */
 static int
 thermal_throttle_cpu_callback(struct notifier_block *nfb,
@@ -271,18 +268,14 @@ thermal_throttle_cpu_callback(struct notifier_block *nfb,
  switch (action) {
  case CPU_UP_PREPARE:
  case CPU_UP_PREPARE_FROZEN:
-  mutex_lock(&therm_cpu_lock);
   err = thermal_throttle_add_dev(dev, cpu);
-  mutex_unlock(&therm_cpu_lock);
   WARN_ON(err);
   break;
  case CPU_UP_CANCELED:
  case CPU_UP_CANCELED_FROZEN:
  case CPU_DEAD:
  case CPU_DEAD_FROZEN:
-  mutex_lock(&therm_cpu_lock);
   thermal_throttle_remove_dev(dev);
-  mutex_unlock(&therm_cpu_lock);
   break;
  }
  return notifier_from_errno(err);
@@ -303,17 +296,11 @@ static __init int thermal_throttle_init_device(void)
 
  cpu_notifier_register_begin();
 
-#ifdef CONFIG_HOTPLUG_CPU
- mutex_lock(&therm_cpu_lock);
-#endif
  /* connect live CPUs to sysfs */
  for_each_online_cpu(cpu) {
   err = thermal_throttle_add_dev(get_cpu_device(cpu), cpu);
   WARN_ON(err);
  }
-#ifdef CONFIG_HOTPLUG_CPU
- mutex_unlock(&therm_cpu_lock);
-#endif
 
  __register_hotcpu_notifier(&thermal_throttle_cpu_notifier);
  cpu_notifier_register_done();
-- 
1.7.1